packages/lesswrong/components/users/NewUserCompleteProfile.tsx

```diff
@@ -58,17 +58,24 @@ const NewUserCompleteProfile: React.FC<NewUserCompleteProfileProps> = ({ classes
     }
     // TODO: Really want them to be able to tell live if their username is
     // taken, but I think that's gonna have to be a later PR.
+    // Note: Probably want to prevent someone from taking an existing
+    // displayName
     // if (usernameIsUnique) ...
     setValidationError('')
   }
   
   async function handleSave() {
     try {
+      if (validationError)return
+      
+      // TODO: loading spinner while running
       await updateUser({variables: {username, subscribeToDigest}})
-      // No errors were thrown, reset form
-      setUsername('')
-      setSubscribeToDigest(false)
     } catch (err) {
+      if (/duplicate key error/.test(err.toString?.())) {
```

Comment:
Why don't you just use `err.message`?

Reply:
I'm not honestly sure if it will have one. It'll have one right? Why does TS think caught errors are `any`?